perf/benchmark/runner/generate_policies/generate_polices.go

```diff
@@ -17,47 +17,81 @@ package main
 import (
     "fmt"
     "bytes"
+    "github.com/ghodss/yaml"
+    "encoding/json"
+
+    authzpb "istio.io/api/security/v1beta1"
 )
 
 type ruleOption struct {
     occurance int
     g         generator
 }
 
-func generateAuthorizationPolicy(action string, ruleToOccurences map[string]*ruleOption) (string, error) {
-    yaml := bytes.Buffer{}
-    yaml.WriteString("spec:\n")
-    yaml.WriteString(" action: " + action + "\n")
+type MyPolicy struct {
+    ApiVersion string `json:"apiVersion"`
+    Kind       string `json:"kind"`
+    Metadata   MetadataStruct `json:"metadata"`
+    Spec       authzpb.AuthorizationPolicy `json:"spec"`
+}
 
-    if (len(ruleToOccurences) == 0) {
-        yaml.WriteString(" {}")
-    } else {
-        yaml.WriteString(" rules:\n")
+type MetadataStruct struct {
+    Name      string  `json:"name"`
+    Namespace string  `json:"namespace"`
+}
+
+func formYaml(policy *MyPolicy) (string, error) {
+    header, err := json.Marshal(policy)
```

Comment:
there might be issues for using json.Marshal on policy because the spec is a protobuf message because some compatibility issues that the reflect package (used by json.Marshal) doesn't work well with protobuf message. And because of this there is actually a special jsonpb.Marshal() just for protobuf message. See https://pkg.go.dev/github.com/golang/protobuf/jsonpb?tab=doc